packages/p2p-v2/src/components/PaymentMethodForm/PaymentMethodForm.tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { TFormState } from 'src/reducers/types';
+import { TSelectedPaymentMethod } from 'types';
+import { p2p } from '@deriv/api';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import CloseCircle from '../../public/ic-close-circle.svg';
+import { ClickableText } from '../ClickableText';
+import { Dropdown } from '../Dropdown';
+import { AddEditPaymentMethodErrorModal, CancelAddPaymentMethodModal, CancelEditPaymentMethodModal } from '../Modals';
+import { PaymentMethodField } from '../PaymentMethodField';
+import { PaymentMethodsHeader } from '../PaymentMethodsHeader';
+import { TextField } from '../TextField';
+import './PaymentMethodForm.scss';
+
+type TPaymentMethodFormProps = {
+    formState: TFormState;
+    onAdd: (selectedPaymentMethod?: TSelectedPaymentMethod) => void;
+    onResetFormState: () => void;
+};
+
+/**
+ * @component This component is used to display the form to add or edit a payment method
+ * @param formState - The current state of the form
+ * @returns {JSX.Element}
+ * @example <PaymentMethodForm formState={formState} />
+ * **/
+const PaymentMethodForm = ({ onAdd, onResetFormState, ...rest }: TPaymentMethodFormProps) => {
+    const {
+        control,
+        formState: { isDirty, isSubmitting, isValid },
+        handleSubmit,
+        reset,
+    } = useForm({ mode: 'all' });
+    const [isOpen, setIsOpen] = useState(false);
```

Comment:
nit: can we name it isModalOpen, just to be a little more specific
```suggestion
    const [isModalOpen, setIsModalOpen] = useState(false);
```